tiledb/sm/fragment/fragment_metadata.h

```diff
@@ -625,6 +625,12 @@ class FragmentMetadata {
   /** Local mutex for thread-safety. */
   std::mutex mtx_;
 
+  /** Mutex per tile offset loading */
+  std::deque<std::mutex> tile_offsets_mtx_;
+
+  /** Mutex per tile var offset loading */
+  std::deque<std::mutex> tile_var_offsets_mtx_;
+
```

Comment:
- What's the motivation for a `deque` instead of a `vector`? We don't push/pop anything from these structures.

- Also add a period at the end of the comments, to be consistent with the nearby member variable block comments.

Reply:
@joe-maley a `std::mutex` isn't moveable so we can't add it to a vector. A deque offers similar guarantees to a vector, more so that a list or other container which we can store a mutex on. I can add this to the comment if you'd like.